packages/framework/src/types/discover.types.ts

```diff
@@ -49,6 +50,18 @@ export type DiscoverStepOutput = {
   options: StepOptions;
 };
 
+export type ChannelPreference = {
+  defaultValue: boolean;
+  readOnly: boolean;
+};
+
+export type DiscoverWorkflowOutputPreference = {
+  workflow: ChannelPreference;
+  channels: {
+    [key in (typeof ChannelTypeEnum)[keyof typeof ChannelTypeEnum]]: ChannelPreference;
```

Comment:
As above, I'd suggest using a typed template literal for better readability

Reply:
See above